### DISCUSSION TITLE
Recursive File Removal

### DISCUSSION THREAD

**Discussion by kishorekaruppusamy** [pyodide/pyodide#4309] [Q&A] [answered]
If i am create a directory inside pyodide environment using `FS.mkdir();` and put some files in it. 
when i try to remove those file when it is not in use i can able to remove file inside a directory using
`FS.unlink('');`
But when i try to remove a directory using `FS.rmdir('');` this i can't able to remove because this method only removes only if the directory is empty.

kindly help me with this to to remove the non empty directory `recursively` instead of removing one by one ..

**Comment by ryanking13** [comment, marked as answer]
`FS` modules come from emscripten, and it seems like [emscripten authors do not want to add a function like `rmtree`](https://github.com/emscripten-core/emscripten/issues/13920) because they are just helper functions that can easily be implemented by end users.

So I think you need to either write your own version of `rmtree` using FS.unlink and FS.rmdir, or call `shutil.rmtree` in Pyodide if it works for you.

**Reply by hoodmane** [reply]
It's funny because they already have `mkdirtree`...

**Comment by kishorekaruppusamy** [comment]
Thanks for the clarification
